packages/eservice-template-process/test/integration/patchUpdateDraftTemplateVersion.test.ts

```diff
@@ -0,0 +1,459 @@
+        mockTemplateVersion.id,
+        {},
+        getMockContextM2MAdmin({
+          organizationId: mockEServiceTemplate.creatorId,
+        })
+      )
+    ).rejects.toThrowError(
+      eserviceTemplateVersionNotFound(
+        eserviceTemplate.id,
+        mockTemplateVersion.id
+      )
+    );
+  });
+
+  it.each(
+    Object.values(eserviceTemplateVersionState).filter(
+      (state) => state !== eserviceTemplateVersionState.draft
+    )
+  )(
+    "should throw notValidEServiceTemplateVersionState if the version is in %s state",
+    async (state) => {
+      const templateVersion: EServiceTemplateVersion = {
+        ...mockTemplateVersion,
+        interface: mockDocument,
+        state,
+      };
+      const eserviceTemplate: EServiceTemplate = {
+        ...mockEServiceTemplate,
+        versions: [templateVersion],
+      };
+      await addOneEServiceTemplate(eserviceTemplate);
+
+      await expect(
+        eserviceTemplateService.patchUpdateDraftTemplateVersion(
+          eserviceTemplate.id,
+          templateVersion.id,
+          {},
+          getMockContextM2MAdmin({ organizationId: eserviceTemplate.creatorId })
+        )
+      ).rejects.toThrowError(
+        notValidEServiceTemplateVersionState(mockTemplateVersion.id, state)
+      );
+    }
+  );
+
+  it("should throw operationForbidden if the requester is not the creator", async () => {
+    const templateVersion: EServiceTemplateVersion = {
+      ...mockTemplateVersion,
+      state: eserviceTemplateVersionState.draft,
+    };
+    const eserviceTemplate: EServiceTemplate = {
+      ...mockEServiceTemplate,
+      versions: [templateVersion],
+    };
+    await addOneEServiceTemplate(eserviceTemplate);
+
+    await expect(
+      eserviceTemplateService.patchUpdateDraftTemplateVersion(
+        eserviceTemplate.id,
+        templateVersion.id,
+        {},
+        getMockContextM2MAdmin({})
+      )
+    ).rejects.toThrowError(operationForbidden);
+  });
+
+  it.each([
+    { dailyCallsPerConsumer: 300, dailyCallsTotal: 200 },
+    { dailyCallsPerConsumer: 300 },
+    { dailyCallsTotal: 50 },
+  ])(
+    "should throw inconsistentDailyCalls if dailyCallsPerConsumer is greater than dailyCallsTotal",
+    async (seed) => {
+      const templateVersion: EServiceTemplateVersion = {
+        ...mockTemplateVersion,
+        state: eserviceTemplateVersionState.draft,
+        dailyCallsPerConsumer: 100,
+        dailyCallsTotal: 200,
+      };
+      const eserviceTemplate: EServiceTemplate = {
+        ...mockEServiceTemplate,
+        versions: [templateVersion],
+      };
+      await addOneEServiceTemplate(eserviceTemplate);
+
+      await expect(
+        eserviceTemplateService.patchUpdateDraftTemplateVersion(
+          eserviceTemplate.id,
+          templateVersion.id,
+          seed,
+          getMockContextM2MAdmin({ organizationId: eserviceTemplate.creatorId })
+        )
+      ).rejects.toThrowError(inconsistentDailyCalls());
+    }
+  );
+
```

Comment:
I would also add this test present for the normal POST update, that tests that everything works when `dailyCallsPerConsumer` is removed/not present: https://github.com/pagopa/interop-be-monorepo/blob/feature/PIN-7685_list-eservice-m2m-events/packages/eservice-template-process/test/integration/updateDraftVersion.test.ts#L240